src/t_stream.c

```diff
@@ -984,9 +984,14 @@ static int streamParseAddOrTrimArgsOrReply(client *c, streamAddTrimArgs *args, i
             /* User didn't provide LIMIT, we must set it. */
 
             if (args->approx_trim) {
-                /* In order to prevent from trimming to do too much work and cause
-                 * latency spikes we limit the amount of work it can do */
+                /* In order to prevent from trimming to do too much work and 
+                 * cause latency spikes we limit the amount of work it can do.
+                 * We have to cap args->limit from both sides in case 
+                 * stream_node_max_entries is 0 or too big (could cause overflow)
+                 */
                 args->limit = 100 * server.stream_node_max_entries; /* Maximum 100 rax nodes. */
+                if (args->limit < 1) args->limit = 1;
+                if (args->limit > 1000000) args->limit = 1000000;
```

Comment:
@guybe7 considering this code (before trimming)
```
        /* Check if we exceeded the amount of work we could do */
        if (limit && (deleted + entries) > limit)
            break;
```
a limit of 1, is almost certainly guaranteed to to do nothing.
maybe we should sample the size of the last node? or have some other big constant?

Reply:
we have two edge case here, and we should need to decide if we want to trim "a lot" or "a little" in each case, and then adjust the code:
1. stream_node_max_entries is very big, so that multiplying it by 100 will overflow: in this case i think we should trim "a lot".
2. stream_node_max_entries is 0: that means the user either wants the node separation to happen by node size, or both node_max configs are 0, which will cause the stream to be stored in one single rax node. i'm not so sure how we should behave in that case, but i think i'm also leaning towards "a lot"..

so maybe it should be
```
if (args->limit <= 0) args->limit = 1000000;
```

@oranagra @Super-long thoughts?

Reply:
i'm ok with that, except maybe the case where stream_node_max_entries is 0 and stream_node_max_bytes is not.
in this case, maybe nodes are rather small, and we don't want "a lot".
on the other hand, this is just a default, a user can always specify the value on his own, so it really doesn't matter much.

Reply:
After all, this is just a default behavior. What we want to ensure is that the default behavior will never lead to unexpected situations for users. I think this is correct.

Reply:
ok. two last arguments:
1. with the current code (two `if`s leading to the some code, we're certainly gonna see some PR soon that suggests to unify them. :stuck_out_tongue_winking_eye:
2. i feel that 1m is too high, how about 1k? not sure if for both, or just for the lower one?

Reply:
mmm the default is 100*100 (max_node_entries) so maybe go with 10000?